src/read_next/read_next.jsx

```diff
@@ -1,10 +1,12 @@
-import React, { useEffect, useState } from "react";
-import "./read_next.css";
+import React, { useEffect, useState } from 'react';
+import './read_next.css';
 import { Link, useParams } from "react-router-dom";
-
+import PaginationDisplay from '../pagination/getData';
 const ReadNext = () => {
     const params = useParams();
-    const [data, setPosts] = useState({ posts: [] });
+    const [data, setPosts] = useState({ post: [] });
```

Comment:
again, if you use array, then name your variables in plural. please pay attention to how you name your variables, this is important
```suggestion
    const [data, setPosts] = useState({ posts: [] });
```